src/test/java/com/jwplayer/southpaw/SouthpawByteArraySetEndToEndTest.java

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright 2018 Longtail Ad Solutions (DBA JW Player)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.jwplayer.southpaw;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.jwplayer.southpaw.json.DenormalizedRecord;
+import com.jwplayer.southpaw.record.BaseRecord;
+import com.jwplayer.southpaw.topic.BaseTopic;
+import com.jwplayer.southpaw.util.ByteArray;
+import com.jwplayer.southpaw.util.FileHelper;
+import com.jwplayer.southpaw.util.KafkaTestServer;
+
+import scala.Tuple2;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.*;
+
+import static org.junit.Assert.*;
+
+
+public class SouthpawByteArraySetEndToEndTest {
+    private static final String CONFIG_PATH = "test-resources/ByteArraySetTest/config.yaml.tmpl";
+    private static final String RELATIONS_PATH = "test-resources/ByteArraySetTest/relation.json";
+    private static final String ROCKSDB_BASE_URI = "file:///tmp/RocksDB/ByteArraySetTest/";
+    private static final String TOPIC_DATA_PATH = "test-resources/ByteArraySetTest/topic/";
+
+    private KafkaTestServer kafkaServer;
+
+    @Before
+    public void setup() throws IOException, URISyntaxException {
+        getRocksDBFolder().delete();
+        getRocksDBFolder().mkdirs();
+        kafkaServer = new KafkaTestServer();
+    }
+
+    @After
+    public void cleanup() throws URISyntaxException {
+        getRocksDBFolder().delete();
+        kafkaServer.shutdown();
+    }
+
+    @Test
+    public void testConsume() throws Exception {
```

Comment:
I added this integration test around the handling of "large" joins by Southpaw to see if I could locate the bug, which was unsuccessful.